### PR TITLE
Add a proper WiFi menu, with Enable/Disable & Disconnect options

### PR DESCRIPTION
**Pull request by davidgfnet**
This also allows WiFi passwords to be remembered. The underlying tool (connman) allows to store passswords (that's why it auto connects whenever you boot a Lakka device), so we expose this so that the user does not have to re-input the pass when connecting to a saved wifi.

Option to forget a password to come in a future PR, for now a password is automatically forgotten if the Wifi connect operation fails (we assume it fails due to password).

Changes the WiFi API to enable these features of course!

### CODE REVIEW

**Comment by twinaphex** [comment]
Excellent stuff once again.

**Comment by twinaphex** [comment]
BTW, for future reference, put wifi code behind HAVE_NETWORKING ifdefs. Some platforms might not define HAVE_NETWORKING at all, so we have to put ifdefs around this WiFi code.

File: menu/menu_displaylist.c [hunk 1]
```diff
@@ -5470,7 +5479,7 @@ unsigned menu_displaylist_build_list(
                   {
                      const char *ssid = scan->net_list[i].ssid;
                      if (menu_entries_append_enum(list,
-                              ssid,
+                              strlen(ssid) ? ssid : msg_hash_to_str(MSG_WIFI_EMPTY_SSID),
```
Comment (jdgleaver) [inline menu/menu_displaylist.c:5482]: I'd use `!string_is_empty(ssid)` here - `strlen()` will produce UB if `ssid` is NULL (even if that can't happen, it's better to be safe!)

File: tasks/task_wifi.c [hunk 2]
```diff
@@ -36,23 +52,64 @@ static void task_wifi_scan_handler(retro_task_t *task)
    task_set_finished(task, true);
 }
 
-bool task_push_wifi_scan(retro_task_callback_t cb)
+static void task_wifi_enable_handler(retro_task_t *task)
```
Comment (jdgleaver) [inline tasks/task_wifi.c:55]: In all these task functions, you need to return early if `task` is `NULL`. This is also true of `task->user_data`, where used.

The existing code was sloppy in this, so I fully understand why you didn't add these checks :)

File: tasks/task_wifi.c [hunk 3]
```diff
@@ -36,23 +52,64 @@ static void task_wifi_scan_handler(retro_task_t *task)
...
    task->type           = TASK_TYPE_BLOCKING;
    task->state          = NULL;
-   task->handler        = task_wifi_scan_handler;
+   task->handler        = task_wifi_connect_handler;
    task->callback       = cb;
-   task->title          = strdup(msg_hash_to_str(
-                           MSG_SCANNING_WIRELESS_NETWORKS));
-
+   task->title          = strdup(msg);
+   task->user_data      = malloc(sizeof(*netinfo));
```
Comment (jdgleaver) [inline tasks/task_wifi.c:104]: Memory leak! `task->user_data` isn't `free()`'d

Also, every `malloc()` and `calloc()` needs to be cast to the correct type, or certain builds will fail

**Comment by jdgleaver** [comment]
Very nice PR! Just a couple of trivial things to tweak :)

**Comment by escalade** [comment]
This PR leads to RetroArch disabling the wifi on startup, the enable wifi option doesn't stick.

**Pull request merged**